packages/core/src/interpreter.ts

```diff
@@ -215,10 +215,16 @@ export class Interpreter<
     // Attach session ID to state
     state._sessionid = this.sessionId;
 
+    // Execute actions
```

Comment:
Here, actions are executed _before_ settling on the next state. If an error occurs, we will transition from the _previous state_ on the `error.*` event.